src/notification.c

```diff
@@ -433,6 +439,36 @@ parse_serialized_icon (GVariantBuilder  *builder,
       else if (xdp_validate_icon (sealed_icon, XDP_ICON_TYPE_NOTIFICATION, NULL, NULL))
         g_variant_builder_add (builder, "{sv}", "icon", icon);
     }
+  else if (strcmp (key, "file-descriptor") == 0)
+    {
+      g_autoptr(GError) local_error = NULL;
+      g_autoptr(XdpSealedFd) sealed_icon = NULL;
+
+      if (!check_value_type (key, value, G_VARIANT_TYPE_HANDLE, error))
+        return FALSE;
+
+      if (!G_IS_UNIX_FD_LIST (fd_list) || g_unix_fd_list_get_length (fd_list) == 0)
```

Comment:
Here you might want to check `fd_list == NULL` instead of using the `G_IS_UNIX_FD_LIST` type check macro